core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -172,7 +172,7 @@ public DeterministicKey(List<ChildNumber> childNumberPath,
     
     /** Clones the key */
     public DeterministicKey(DeterministicKey keyToClone, DeterministicKey newParent) {
-        super(keyToClone.priv, keyToClone.pub.get(), keyToClone.pub.isCompressed());
+        super(keyToClone.priv, keyToClone.pub);
```

Comment:
I'm not sure what is going on here. Partly, I'm confused by the `/** Clones the key */` comment which is not clear to me which key is being cloned. And with the change there is no cloning going on.

Reply:
> I think the comment on the "clone constructor" should be changed and/or clarified.

I have commits about the clone constructor coming up. It should be made a static `withSomething()`, and the "something" is also not entirely clear to me but we'll figure this out.

However, this commit is not about the clone constructor. It's about re-using of LazyECPoint, rather than disassembling it into components and then reassembling them in the called constructor. I think because LazyECPoint is now immutable, we can safely re-use it and it will make our code cleaner.